src/utils/dom-utils/addDescriptionGenerator.ts

```diff
@@ -23,11 +21,15 @@ const injectDescriptionGeneratorButton = async () => {
         }
     }
 
-    const selector = isGithubPullRequestPage(window.location.href) ? GITHUB_PR_COMMENT_EDITOR_SELECTOR : GITHUB_NEW_PR_COMMENT_EDITOR_SELECTOR;
-    const commentFormatRow = document.getElementsByClassName(selector)[0];
+    const prActionsClassName = "ActionBar-item-container";
+    const actionBars = document.getElementsByClassName(prActionsClassName);
+    const actionBarsArray = Array.from(actionBars);
     const addGeneratorButton = DescriptionGeneratorButton();
+    const firstPrDescription = actionBarsArray[0];
 
-    commentFormatRow.insertBefore(addGeneratorButton, commentFormatRow.firstChild);
+    if (!firstPrDescription.querySelector("#ai-description-button")) {
+        firstPrDescription.insertBefore(addGeneratorButton, firstPrDescription.firstChild);
+    }
```

Comment:
refactor (blocking): Since the first element from `document.getElementsByClassName(prActionsClassName);` is being retrieved, it's the same thing as doing `document.querySelector(".ActionBar-item-container")` as querySelector grabs the first element in the document that matches the specified selector.

Also, `const addGeneratorButton = DescriptionGeneratorButton();` can be moved into the if block as that's the only time it's used.

I haven't tested it, but this should result in the same outcome.

```suggestion
    const firstPrDescription = document.querySelector(".ActionBar-item-container");

    if (firstPrDescription && !firstPrDescription.querySelector("#ai-description-button")) {
        const addGeneratorButton = DescriptionGeneratorButton();

        firstPrDescription.insertBefore(addGeneratorButton, firstPrDescription.firstChild);
    }
```

Reply:
Thank you for the suggestions, I did not think about it and moved with the original code. I think I will need to use the queryselector all as well in this PR: https://github.com/open-sauced/ai/pull/285